### PR TITLE
refactor(es/transforms/compat): visit_mut for destructuring

### PR DESCRIPTION
**Pull request by kwonoj**
One more check off for https://github.com/swc-project/swc/issues/2370, refactor destructuring into visit_mut.

I tried to make straightforward refactor as same as before.

### CODE REVIEW

File: crates/swc_ecma_transforms_compat/src/macros.rs [hunk 1]
```diff
@@ -168,3 +168,159 @@ macro_rules! impl_fold_fn {
         }
     };
 }
+
+macro_rules! impl_visit_mut_fn {
```
Comment (kwonoj) [inline crates/swc_ecma_transforms_compat/src/macros.rs:172]: There are few places still using `impl_fold_fn` macro, so PR do not delete it yet.

File: crates/swc_ecma_transforms_compat/src/es2015/destructuring.rs [hunk 2]
```diff
@@ -518,22 +535,27 @@ impl Destructuring {
...
-            .chain(body.stmts)
-            .collect()
+            let mut stmt = Stmt::Decl(Decl::Var(VarDecl {
+                span: DUMMY_SP,
+                kind: VarDeclKind::Let,
+                decls,
+                declare: false,
+            }));
+
+            stmt.visit_mut_children_with(self);
```
Comment (kdy1) [inline crates/swc_ecma_transforms_compat/src/es2015/destructuring.rs:547]: I think this is bit different from original, as original one is `.fold_with()`.
So it should be `visit_mut_with(self)` to be identical.

Comment (kdy1) [reply crates/swc_ecma_transforms_compat/src/es2015/destructuring.rs:547]: But well, as there's no `visit_mut_stmt` so it does not matter.

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**